journal.py

```diff
@@ -148,77 +177,50 @@ def markd(input):
     return markdown.markdown(input, extension=["CodeHilite"])
 
 
-def write_entry(request):
-    """write a single entry to the database"""
-    title = request.params.get('title', None)
-    text = request.params.get('text', None)
-    created = datetime.datetime.utcnow()
-    request.db.cursor().execute(INSERT_ENTRY, [title, text, created])
+# def write_entry(request):
+#     """write a single entry to the database"""
+#     title = request.params.get('title', None)
+#     text = request.params.get('text', None)
+#     created = datetime.datetime.utcnow()
+#     request.db.cursor().execute(INSERT_ENTRY, [title, text, created])
 
 
 @view_config(route_name='home', renderer='templates/list.jinja2')
 def read_entries(request):
     """return a list of all entries as dicts"""
-    cursor = request.db.cursor()
-    cursor.execute(DB_ENTRIES_LIST)
-    keys = ('id', 'title', 'text', 'created')
-    entries = [dict(zip(keys, row)) for row in cursor.fetchall()]
-    for entry in entries:
-        entry['text'] = markdown.markdown(entry['text'], extensions=['codehilite', 'fenced_code'])
+    entries = Entry.all()
     return {'entries': entries}
 
 
 @view_config(route_name='detail', renderer='templates/detail.jinja2')
 def detail_entry(request):
-    """return a single entry"""
-    cursor = request.db.cursor()
-    cursor.execute(SELECT_ID, (request.matchdict['id'], ))
-    keys = ('id', 'title', 'text', 'created')
-    row = cursor.fetchone()
-    entry = dict(zip(keys, row))
-    entry['text'] = markdown.markdown(
-        entry['text'], extensions=['codehilite', 'fenced_code'])
+    """return a single entry by its id"""
+    entry = Entry.by_id(request.matchdict['id'])
```

Comment:
Might still be safer here to use a `get` on the matchdict.  Or catch the key error and return an `HTTPNotFound`